hazelcast-tpc-engine/src/test/java/com/hazelcast/internal/tpc/EventloopTest.java

```diff
@@ -21,28 +21,49 @@
 import org.junit.Test;
 
 import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
 
 import static com.hazelcast.internal.tpc.TpcTestSupport.assertEqualsEventually;
+import static com.hazelcast.internal.tpc.TpcTestSupport.assertTrueEventually;
 import static com.hazelcast.internal.tpc.TpcTestSupport.terminate;
 import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.Assert.assertEquals;
 
 public abstract class EventloopTest {
 
     private Reactor reactor;
 
-    public abstract Reactor newReactor();
+    public abstract ReactorBuilder newReactorBuilder();
 
     @Before
     public void before() {
-        reactor = newReactor();
-        reactor.start();
+        ReactorBuilder reactorBuilder = newReactorBuilder();
+        reactor = reactorBuilder.build().start();
     }
 
     @After
     public void after() {
         terminate(reactor);
     }
 
+    @Test
+    public void test_schedule() {
+        Task task = new Task();
+
+        reactor.offer(() -> reactor.eventloop.schedule(task, 1, SECONDS));
+
+        assertTrueEventually(() -> assertEquals(1, task.count.get()));
+    }
```

Comment:
How is this test related to the changes in the production code? Without updating cached time were there any issues with scheduling?

Reply:
The tests ran for a very long time. I'll add more scheduling tests, but testing everything with time is a PITA because either you set the timing too low and you get spurious failures, or you set it too high and tests become too slow.

Reply:
I see, can you extract the test into separate PR though? I think it's (commit message vs what is changed inside the commit) confusing as it is. I will directly approve the other PR.

Reply:
I don't see the need for it. The test exposed the problem.